src/list.c

```diff
@@ -68,3 +107,122 @@ int list_add(struct list_t *list, struct entry_t *entry) {
     previous_node->next = new_node;
```

Comment:
Just noticed that if the list only has a head, it will never go into the `while (node->next != 0)` loop, leaving `previous_node` undefined...

Reply:
Actually, 104-107 line don't work (I forgot to check this part last time, oops), I suppose it's supposed to add the entry at the end of the list? if that's the case it should probably be
```
new_node->entry=entry;
new_node->next=NULL; //since it's the end of the list?
node->next = new_node
```